apps/wizard/app_pages/chart_to_gif.py

```diff
@@ -0,0 +1,150 @@
+    st.session_state.images_folder = DOWNLOADS_DIR
+if "image_paths" not in st.session_state:
+    st.session_state.image_paths = None
+if "gif_file" not in st.session_state:
+    st.session_state.gif_file = None
+if "iframe_html" not in st.session_state:
+    st.session_state.iframe_html = None
+if "years" not in st.session_state:
+    # NOTE: The range of years will be loaded automatically from the chart's metadata. We just define this range here to avoid typing issues.
+    st.session_state.years = range(2000, 2022)
+
+# Step 1: Input chart URL and get years.
+st.header("Step 1: Chart URL and timeline")
+chart_url = st.text_input("Enter grapher URL", "")
+
+# Get slug from URL.
+slug = get_chart_slug(chart_url=chart_url)
+
+# Set images folder and output file.
+st.session_state.images_folder = DOWNLOADS_DIR / slug
+st.session_state.gif_file = DOWNLOADS_DIR / f"{slug}.gif"
+
+
+if st.button("Get chart"):
+    # Embed the iframe in the app.
+    st.session_state.iframe_html = f"""
+    <iframe src="{chart_url}" loading="lazy"
+            style="width: 100%; height: 600px; border: 0px none;"
+            allow="web-share; clipboard-write"></iframe>
+    """
+    st.session_state.years = get_years_in_chart(chart_url)
+    st.session_state.show_image_settings = True
+
+# Display iframe if it was fetched.
+if st.session_state.iframe_html:
+    st.markdown("Modify chart as you wish, click on share-> copy link, and paste it in the box above.")
+    st.components.v1.html(st.session_state.iframe_html, height=600)  # type: ignore
+
+if st.session_state.get("show_image_settings"):
+    st.header("Step 2: Image generation settings")
+    # Slider for year range.
+    year_min, year_max = st.slider(
+        "Select year range",
+        min_value=min(st.session_state.years),
+        max_value=max(st.session_state.years),
+        value=(min(st.session_state.years), max(st.session_state.years)),
+    )
+
+    # Tab and year range settings.
+    tab = st.radio("Select tab", ["map", "chart"])
+    year_range_open = st.checkbox(
+        "Year range open",
+        value=True,
+        help="Only relevant for the chart view. If checked, the year range will be open. Uncheck if you want to generate a sequence of bar charts.",
+    )
+
+    # Button to generate images.
+    # TODO: It's convenient to keep images in the folder, to avoid downloading them again.
+    #  However, we should detect if the images are already there and have a button to delete them (in case the settings for image generation change).
+    if st.button("Get images"):
+        years = range(year_min, year_max + 1)
+        st.session_state.image_paths = get_images_from_chart_url(
+            chart_url=chart_url,
+            png_folder=st.session_state.images_folder,
+            tab=tab,
+            years=years,
+            year_range_open=year_range_open,
+            max_workers=None,
+            max_num_years=100,
+        )
+        st.session_state.show_gif_settings = True
```

Comment:
As a user, I'm a bit confused by this step. Maybe this can just run after submitting step 1?

I think the user can just grab the link, including its query params, from admin and that's it.

If there really the need to have the options "year range open", "year range" etc. I'd suggest just merging these options with step 3.

Less steps = simpler

Reply:
Yes, this could be simplified a bit, but there are some complications. For example, the grapher URL doesn't tell you what the default chart view is (map or chart), or the range of years. We could get all this from DB, but I'll keep it as-is for now, we can improve it in a separate PR.

Reply:
in that case, you can just move all these parameters to step 3 (to be renamed 2), and just have two steps.

1) define the chart slug / URL
2) generate gif (by getting images, etc.)

Reply:
I think it's good to keep "Get images" explicit. Otherwise you may attempt to get all images automatically from a chart that has many years. It makes sense for now to keep 3 steps.

Reply:
mmh, I see. FWIW, it ran quite fast for me.

You could maybe set a threshold, and show a year-filtering option IF the range of years is above X. Otherwise, you could run it without worrying.

I think it'd make the experience smoother

Reply:
@pabloarosado unresolving just to make it visible again.

Reply:
Now it's just two steps (I've refactored things a bit), what do you think?